apps/src/StudioApp.js

```diff
@@ -914,6 +917,20 @@ StudioApp.prototype.addChangeHandler = function(newHandler) {
 };
 
 StudioApp.prototype.runChangeHandlers = function() {
+  if (
+    this.isRunning() &&
+    this.editDuringRunAlert === undefined &&
+    (this.isUsingBlockly() ||
+      this.getCode().trim() !== this.executingCode.trim())
```

Comment:
Given that this is a complicated if statement, consider adding a comment explaining what's going on here.